sdk/storage_datalake/src/shared_key_authorization_policy.rs

```diff
@@ -0,0 +1,230 @@
+use azure_core::{Context, Policy, PolicyResult, Request, Response};
+use http::header::AUTHORIZATION;
+use http::{HeaderMap, HeaderValue, Method};
+use ring::hmac;
+use std::sync::Arc;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SharedKeyAuthorizationPolicy {
```

Comment:
Should be `StorageSharedKeyCredential` that is public, which takes `account_name` then `account_key`. The policy itself shouldn't need to be public (at least, not exported from the crate but perhaps just from this module). The associated builder should then take the `StorageSharedKeyCredential` as an option for authenticating.

Reply:
Thanks, this is definitely the longer term goal. Short term, I'm just switching from Bearer token to Shared key in the quickest way possible to unblock another project.

Reply:
@heaths please take another look now. I introduced the StorageSharedKeyCredential struct.